ui/src/lib/yjs/useYWorkflow.ts

```diff
@@ -42,12 +42,53 @@ export default ({
   );
 
   const handleWorkflowAdd = useCallback(
-    (position?: XYPosition) =>
+    (position?: XYPosition, nodes?: Node[], edges?: Edge[]) =>
```

Comment:
Adding nodes and edges to here is a little confusing since most places `handleWorkflowAdd` is used only wants position.
Consider a separate function like `handleWorkflowAddFromSelection` or something, while keeping in mind keeping duplicated logic to a minimum.